### PR TITLE
o/snapstate: add AffectedByRefreshCandidates helper

### PR DESCRIPTION
**Pull request by stolowski**
Add AffectedByRefreshCandidates helper for use from hookstate. Having the helper avoids exporting of snapstate.refreshCandidate (there is a simplified version of refreshCandidate in hookstate, for gating hook only).

### CODE REVIEW

File: overlord/snapstate/autorefresh_gating.go [hunk 1]
```diff
@@ -380,6 +380,22 @@ type AffectedSnapInfo struct {
 	AffectingSnaps map[string]bool
 }
 
+// AffectedByRefreshCandidates returns information about all snaps affected by
+// current refresh-candidates in the state.
+func AffectedByRefreshCandidates(st *state.State) (map[string]*AffectedSnapInfo, error) {
+	var candidates map[string]*refreshCandidate
```
Comment (pedronis) [inline overlord/snapstate/autorefresh_gating.go:386]: as we care only about the keys this can use *json.RawMessage

**Review by pedronis — APPROVED** [review]
small suggestion

File: overlord/snapstate/autorefresh_gating_test.go [hunk 2]
```diff
@@ -1115,6 +1115,32 @@ func (s *autorefreshGatingSuite) TestCreateAutoRefreshGateHooks(c *C) {
...
+
+	candidates := map[string]*snapstate.RefreshCandidate{"snap-a": {}}
+	st.Set("refresh-candidates", &candidates)
+
+	affected, err = snapstate.AffectedByRefreshCandidates(st)
+	c.Assert(err, IsNil)
+	c.Check(affected, DeepEquals, map[string]*snapstate.AffectedSnapInfo{
+		"snap-a": {
+			AffectingSnaps: map[string]bool{
+				"snap-a": true,
```
Comment (mvo5) [inline overlord/snapstate/autorefresh_gating_test.go:1140]: I'm not super familiar with how this is supposed to work, but should we have self-references like this?

**Review by mvo5 — APPROVED** [review]
LGTM

**Pull request merged**